concordium-std-derive/src/lib.rs

```diff
@@ -1210,10 +1254,48 @@ fn schema_type_field_type(field: &syn::Field) -> syn::Result<proc_macro2::TokenS
 fn schema_type_fields(fields: &syn::Fields) -> syn::Result<proc_macro2::TokenStream> {
     match fields {
         syn::Fields::Named(_) => {
+            let mut used_field_names = HashMap::new();
             let fields_tokens: Vec<_> = fields
                 .iter()
                 .map(|field| {
-                    let field_name = field.ident.clone().unwrap().to_string(); // safe since named fields
+                    let (field_name, field_span) = if let Some(renamed_field_name) =
+                        find_field_attribute_value(&field.attrs, "rename")?
+                    {
+                        match renamed_field_name {
+                            syn::Lit::Str(renamed_field_name) => {
+                                (renamed_field_name.value(), renamed_field_name.span())
+                            }
+                            _ => {
+                                return Err(syn::Error::new(
+                                    renamed_field_name.span(),
+                                    "Rename attribute value must be a string.",
+                                ))
+                            }
+                        }
+                    } else {
+                        (field.ident.clone().unwrap().to_string(), field.ident.span())
+                        // safe since
+                        // // named fields
+                    };
+
+                    // Check whether the name is defined multiple times
+                    if let Some(used_field_span) =
+                        used_field_names.insert(field_name.clone(), field_span)
+                    {
+                        let mut error_at_first_def = syn::Error::new(
+                            used_field_span,
+                            format!("the name `{}` is defined multiple times", field_name),
+                        );
+                        let error_at_second_def = syn::Error::new(
+                            field_span,
+                            format!("the name `{}` is defined multiple times", field_name),
+                        );
+                        // Combine the errors to show both at once
+                        error_at_first_def.combine(error_at_second_def);
+
+                        return Err(error_at_first_def);
+                    }
```

Comment:
The same goes for this part for checking duplicates. The error messages look the same.